includes/class-signature.php

```diff
@@ -194,7 +81,7 @@ protected static function check_legacy_key_pair_for( $user_id ) {
 	 */
 	public static function generate_signature( $user_id, $http_method, $url, $date, $digest = null ) {
```

Comment:
it would be awesome to pass the public key here instead of the `$user_id`, to be completely user agnostic.